gap/weights.gi

```diff
@@ -780,3 +780,52 @@ function(digraph, start, destination)
 
   return flows;
 end);
+
+#############################################################################
+# 6. Random edge weighted digraphs
+#############################################################################
+
+DIGRAPHS_RandomUniqueWeights := function(digraph)
+  local outs, weightsSource;
+  outs := OutNeighbours(digraph);
+
+  # Unique weights are taken randomly from [1..nredges]
+  weightsSource := Shuffle([1 .. DigraphNrEdges(digraph)]);
+
+  # Assign the weights to the edges
+  return List(DigraphVertices(digraph),
+              u -> List(outs[u], _ -> Remove(weightsSource)));
```

Comment:
Interesting, does this work? If so, then nice!

Reply:
Yep, I was pleased with how concise I'd got that. `Remove` acts as a pop function, which allows for this one-liner.